addons/controllers/addon_controller.go

```diff
@@ -115,7 +115,19 @@ func (r *AddonReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ct
 		return ctrl.Result{}, err
 	}
 
-	tkrName := util.GetTKRNameForCluster(ctx, r.Client, cluster)
+	// "classy" UTKG clusters have "run.tanzu.vmware.com/tkr" label or
+	// are not labeled with any TKR related labels
+	// (in case of not having run.tanzu.vmware.com/resolve-tkr annotation).
+	// These clusters need to get reconciled by the cluster bootstrap controller.
+	// only legacy-style clusters are labeled with "tanzuKubernetesRelease"
+	// and should be reconciled by the addons controller.
+	for k := range cluster.Labels {
+		if k == constants.TKRLabelClassyClusters {
+			return ctrl.Result{}, nil
+		}
+	}
+
+	tkrName := util.GetTKRNameForCluster(ctx, r.Client, cluster, constants.TKRLabel)
```

Comment:
Could we add the check to WithEventFilter? https://github.com/vmware-tanzu/tanzu-framework/blob/d5687c28ead83902696943eed40aeaf0efced0c5/addons/controllers/addon_controller.go#L90

Reply:
@vijaykatam The label only exists in the cluster object and not other resources we watch. For that reason, can't add the check to ```WithEventFilter``` as the resources are not converted to cluster object yet. Please correct me otherwise.

Reply:
Please check how `clusterApiPredicates.ResourceNotPaused` works. It checks for annotations. You would be checking for labels.

Reply:
I've looked at ```clusterApiPredicates.ResourceNotPaused```. The pause annotation can exist for all resources we watch (cluster, secret, KubeadmControlPlane, configmap, tkr). But the tkr label only exists for the cluster object out of those. That's why I'm afraid adding the predicate for tkr labels in WithEventFilter would not address our concern.

Reply:
Could you check object meta for Cluster Kind so that it ignores other types? I am confused why the checks need to be duplicated if it is already in the event filter.

Reply:
Yes I'll check for the type. For the cases where we convert other kinds to cluster (https://github.com/vmware-tanzu/tanzu-framework/blob/main/addons/controllers/addon_controller.go#L63) for example ```AddonSecretToClusters```,  which they call ```EnqueueRequestsFromMapFunc``` directly after getting converted, will the same high level ```WithEventFilter```  be sufficient ? Please let me know if I was not clear